server/auth.js

```diff
@@ -0,0 +1,24 @@
+var admin = require("firebase-admin");
+admin.initializeApp({
+    projectId: "daily-focus-a7423",
+});
+function authorise(req, res, next) {
+    let idToken;
+    if (req.headers.authorization && req.headers.authorization.startsWith("Bearer ")) {
+        idToken = req.headers.authorization.split("Bearer ")[1];
+    } else {
+        return res.status(401).json({ message: "Unauthorized access" });
+    }
+
+    admin
+        .auth()
+        .verifyIdToken(idToken)
+        .then((token) => {
+            const UUID = token.user_id;
```

Comment:
If this is the actual user ID, why not just return it so the endpoints can use the id when querying the database?